packages/twenty-server/src/engine/core-modules/workspace/dtos/update-workspace-input.ts

```diff
@@ -14,8 +14,91 @@ export class UpdateWorkspaceInput {
   @Field({ nullable: true })
   @IsString()
   @IsOptional()
-  @Matches(/^[a-z0-9][a-z0-9-]{1,28}[a-z0-9]$/)
-  @ForbiddenWords(['demo'])
+  @Matches(/^(?!api-).*^[a-z0-9][a-z0-9-]{1,28}[a-z0-9]$/)
+  @NotIn([
```

Comment:
IsNotIn from class-validator should do exactly the same I think, you won't need to implement your own validator

Reply:
The custom validator support regex unlike the default one. But I moved the validation of "api-*" in the Matches validator. So I will remove it.